web/main.ts

```diff
@@ -2324,15 +2358,30 @@ class GitGraphView {
 
 	/* Commit Details View */
 
-	public loadCommitDetails(commitElem: HTMLElement) {
+	public loadCommitDetails(commitElem: HTMLElement, parentIndex: number) {
 		const commit = this.getCommitOfElem(commitElem);
-		if (commit === null) return;
+		if (commit === null) {
+			this.showErrorOnNonLoadedCommit();
+			return;
+		}
 
 		this.closeCommitDetails(false);
 		this.saveExpandedCommitLoading(parseInt(commitElem.dataset.id!), commit.hash, commitElem, null, null);
 		commitElem.classList.add(CLASS_COMMIT_DETAILS_OPEN);
 		this.renderCommitDetailsView(false);
-		this.requestCommitDetails(commit.hash, false);
+		this.requestCommitDetails(commit.hash, false, parentIndex);
+	}
+
+	private showErrorOnNonLoadedCommit() {
+		const actionName = this.moreCommitsAvailable ? 'Load More Commits' : null;
+
+		dialog.showError('The commit could not be found in the loaded commits.<br>' +
+			'This can happen for several reasons:' +
+			'<ol>' +

```

Comment:
* I think this message is a great addition, and makes it much clearer to the user. Great work!
* I found that styling the list (e.g. left align, margins etc.) made it more readable, please do this.
* You can make some of the lines conditionally visible, based on the current view state:
    * If `this.moreCommitsAvailable` is `false`, then line 2381 isn't necessary.
    * If `getIncludeCommitsMentionedByReflogs(this.gitRepos[this.currentRepo].includeCommitsMentionedByReflogs)` is `true`, then line 2383 isn't necessary.

Reply:
I've implemented your suggestions. Do you think we should also check if filtering is applied? If so how should I go about it? 

I see there's a list of branches in `this.currentBranches`, and I found the following condition on line 243:
`this.currentBranches !== null && !(this.currentBranches.length === 1 && this.currentBranches[0] === SHOW_ALL_BRANCHES`, that may test exactly for this.

Did I get it right?

Reply:
That is correct for the branch filters, however there are other types of filters that the user may have applied (which make it more complicated to test). I think it’s fine to always show the line about filters, as it’s a good general statement to have.